cubeviz/controls/flux_unit_registry.py

```diff
@@ -0,0 +1,289 @@
+import os
+import yaml
+
+from astropy import units as u
+
+DEFAULT_FLUX_UNITS_CONFIGS = os.path.join(os.path.dirname(__file__), 'registered_flux_units.yaml')
+
+
+def _add_unit_to_list(unit_list, target_unit):
+    """
+    Given a list of units, add target units
+    after checking for duplicates
+    :param unit_list: list of units or string
+    :param target_unit: unit or string
+    :return: updated unit
+    """
+
+    if isinstance(target_unit, str):
+        current_unit = u.Unit(target_unit)
+        string_unit = target_unit
+    else:
+        current_unit = target_unit
+        string_unit = target_unit.to_string()
+
+    duplicate = False
```

Comment:
I think something like:
```
unit_list = list(set(unit_list + [string_unit]))
```
is shorter and reasonably clear.  My stuff above can replace your for loop and if not duplicate part.

Reply:
set is a beautiful thing...

Reply:
unfortunately order matters here 😢